Implementations of `javax.wsdl.extensions.ExtensibilityElement` need to get registered for reflection

Reproducer app: https://github.com/ppalaga/d1882-native-extensibility/tree/main

Run 

```
mvn clean verify -Pnative -Dquarkus.native.container-build
...
[INFO] --- failsafe:3.3.1:integration-test (default) @ d1882-native-extensibility ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] [stdout] 2025-08-22 14:35:44,438 WARN  [io.qua.run.log.LoggingSetupRecorder] (main) Log level TRACE for category 'org.apache.cxf' set below minimum logging level DEBUG, promoting it to DEBUG. Set the build time configuration property 'quarkus.log.category."org.apache.cxf".min-level' to 'TRACE' to avoid this warning
[INFO] Running io.quarkiverse.cxf.reproducer.it.ClientesServiceSoapIT
[INFO] [stdout] 2025-08-22 14:35:45,720 WARN  [io.qua.run.log.LoggingSetupRecorder] (main) Log level TRACE for category 'org.apache.cxf' set below minimum logging level DEBUG, promoting it to DEBUG. Set the build time configuration property 'quarkus.log.category."org.apache.cxf".min-level' to 'TRACE' to avoid this warning
[INFO] [stdout] Executing "/home/ppalaga/orgs/cxf/reproducers/d1882-native-extensibility/target/d1882-native-extensibility-1.0.0-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/home/ppalaga/orgs/cxf/reproducers/d1882-native-extensibility/target/quarkus.log -Dquarkus.log.file.enabled=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.profile=prod -Dquarkus.native.container-build=true"
[INFO] [stdout] __  ____  __  _____   ___  __ ____  ______ 
[INFO] [stdout]  --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
[INFO] [stdout]  -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
[INFO] [stdout] --\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
[INFO] [stdout] 2025-08-22 14:35:45,767 WARN  [io.qua.run.log.LoggingSetupRecorder] (main) Log level TRACE for category 'org.apache.cxf' set below minimum logging level DEBUG, promoting it to DEBUG. Set the build time configuration property 'quarkus.log.category."org.apache.cxf".min-level' to 'TRACE' to avoid this warning
[INFO] [stdout] 2025-08-22 14:35:45,768 DEBUG [org.apa.cxf.res.DefaultResourceManager] (main) resolving resource <org.apache.cxf.io.DelayedCachedOutputStreamCleaner/bus> type <interface org.apache.cxf.Bus>
[INFO] [stdout] 2025-08-22 14:35:45,768 DEBUG [org.apa.cxf.res.DefaultResourceManager] (main) resolving resource <null> type <interface org.apache.cxf.Bus>
[INFO] [stdout] 2025-08-22 14:35:45,770 FINE  [org.apa.cxf.wsd.ser.fac.ReflectionServiceFactoryBean] (main) Creating Service {http://intranet.cliente.com/services/}ClientesService from WSDL: wsdl/ClientesService.wsdl
[INFO] [stdout] 2025-08-22 14:35:45,776 ERROR [io.qua.run.Application] (main) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
[INFO] [stdout]         at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
[INFO] [stdout]         at io.quarkus.runtime.Application.start(Application.java:101)
[INFO] [stdout]         at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
[INFO] [stdout]         at io.quarkus.runtime.Quarkus.run(Quarkus.java:80)
[INFO] [stdout]         at io.quarkus.runtime.Quarkus.run(Quarkus.java:51)
[INFO] [stdout]         at io.quarkus.runtime.Quarkus.run(Quarkus.java:144)
[INFO] [stdout]         at io.quarkus.runner.GeneratedMain.main(Unknown Source)
[INFO] [stdout] Caused by: org.apache.cxf.service.factory.ServiceConstructionException: SERVICE_CREATION_MSG
[INFO] [stdout]         at org.apache.cxf.wsdl11.WSDLServiceFactory.<init>(WSDLServiceFactory.java:87)
[INFO] [stdout]         at org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean.buildServiceFromWSDL(ReflectionServiceFactoryBean.java:393)
[INFO] [stdout]         at org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean.initializeServiceModel(ReflectionServiceFactoryBean.java:527)
[INFO] [stdout]         at org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean.create(ReflectionServiceFactoryBean.java:262)
[INFO] [stdout]         at org.apache.cxf.jaxws.support.JaxWsServiceFactoryBean.create(JaxWsServiceFactoryBean.java:199)
[INFO] [stdout]         at org.apache.cxf.frontend.AbstractWSDLBasedEndpointFactory.createEndpoint(AbstractWSDLBasedEndpointFactory.java:103)
[INFO] [stdout]         at org.apache.cxf.frontend.ServerFactoryBean.create(ServerFactoryBean.java:168)
[INFO] [stdout]         at org.apache.cxf.jaxws.JaxWsServerFactoryBean.create(JaxWsServerFactoryBean.java:210)
[INFO] [stdout]         at io.quarkiverse.cxf.transport.CxfHandler.<init>(CxfHandler.java:155)
[INFO] [stdout]         at io.quarkiverse.cxf.CXFRecorder.initServer(CXFRecorder.java:275)
[INFO] [stdout]         at io.quarkus.runner.recorded.CxfEndpointImplementationProcessor$startRoute912873805.deploy_0(Unknown Source)
[INFO] [stdout]         at io.quarkus.runner.recorded.CxfEndpointImplementationProcessor$startRoute912873805.deploy(Unknown Source)
[INFO] [stdout]         ... 7 more
[INFO] [stdout] Caused by: javax.wsdl.WSDLException: WSDLException (at /wsdl:definitions/wsdl:binding[2]/http:binding): faultCode=CONFIGURATION_ERROR: Problem instantiating Java extensionType 'com.ibm.wsdl.extensions.http.HTTPBindingImpl'.: java.lang.InstantiationException: com.ibm.wsdl.extensions.http.HTTPBindingImpl
[INFO] [stdout]         at javax.wsdl.extensions.ExtensionRegistry.createExtension(ExtensionRegistry.java:383)
[INFO] [stdout]         at com.ibm.wsdl.extensions.http.HTTPBindingSerializer.unmarshall(HTTPBindingSerializer.java:69)
[INFO] [stdout]         at com.ibm.wsdl.xml.WSDLReaderImpl.parseExtensibilityElement(WSDLReaderImpl.java:1861)
[INFO] [stdout]         at com.ibm.wsdl.xml.WSDLReaderImpl.parseBinding(WSDLReaderImpl.java:988)
[INFO] [stdout]         at com.ibm.wsdl.xml.WSDLReaderImpl.parseDefinitions(WSDLReaderImpl.java:332)
[INFO] [stdout]         at com.ibm.wsdl.xml.WSDLReaderImpl.readWSDL(WSDLReaderImpl.java:2352)
[INFO] [stdout]         at com.ibm.wsdl.xml.WSDLReaderImpl.readWSDL(WSDLReaderImpl.java:2338)
[INFO] [stdout]         at io.quarkiverse.cxf.wsdl.QuarkusWSDLManager.loadDefinition(QuarkusWSDLManager.java:141)
[INFO] [stdout]         at org.apache.cxf.wsdl11.WSDLManagerImpl.getDefinition(WSDLManagerImpl.java:165)
[INFO] [stdout]         at org.apache.cxf.wsdl11.WSDLServiceFactory.<init>(WSDLServiceFactory.java:85)
[INFO] [stdout]         ... 18 more
[INFO] [stdout] Caused by: java.lang.InstantiationException: com.ibm.wsdl.extensions.http.HTTPBindingImpl
[INFO] [stdout]         at java.base@21.0.8/java.lang.Class.newInstance(DynamicHub.java:719)
[INFO] [stdout]         at javax.wsdl.extensions.ExtensionRegistry.createExtension(ExtensionRegistry.java:367)
[INFO] [stdout]         ... 27 more
[INFO] [stdout] Caused by: java.lang.NoSuchMethodException: com.ibm.wsdl.extensions.http.HTTPBindingImpl.<init>()
[INFO] [stdout]         at java.base@21.0.8/java.lang.Class.checkMethod(DynamicHub.java:1078)
[INFO] [stdout]         at java.base@21.0.8/java.lang.Class.getConstructor0(DynamicHub.java:1241)
[INFO] [stdout]         at java.base@21.0.8/java.lang.Class.newInstance(DynamicHub.java:706)
[INFO] [stdout]         ... 28 more
```

--- a/docs/modules/ROOT/examples/client-server/application.properties
+++ b/docs/modules/ROOT/examples/client-server/application.properties
@@ -218,4 +218,13 @@ quarkus.cxf.client.javaNetSslClient.client-endpoint-url = https://localhost:${qu
 quarkus.cxf.client.javaNetSslClient.service-interface = io.quarkiverse.cxf.it.HelloService
 quarkus.cxf.client.javaNetSslClient.logging.enabled = true
 
+# ExtensorsTest
+quarkus.cxf.codegen.wsdl2java.extensors.includes = wsdl/ExtensorsService.wsdl
+quarkus.cxf.codegen.wsdl2java.extensors.package-names = io.quarkiverse.cxf.it.server.extensors.model
+quarkus.cxf.codegen.wsdl2java.extensors.wsdl-location = classpath:wsdl/ExtensorsService.wsdl
+quarkus.cxf.codegen.wsdl2java.extensors.output-directory = target/generated-sources/wsdl2java-extensors
+
+quarkus.cxf.client.extensors.wsdl=wsdl/ExtensorsService.wsdl
+quarkus.cxf.client.extensors.client-endpoint-url=http://localhost:${quarkus.http.test-port}/soap/ExtensorsService
+
 quarkus.default-locale = en_US

--- a/extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/Wsdl4jProcessor.java
+++ b/extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/Wsdl4jProcessor.java
@@ -4,10 +4,18 @@ import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Set;
+import java.util.stream.Collectors;
 import java.util.stream.Stream;
+
+import javax.wsdl.extensions.ExtensibilityElement;
+
+import org.jboss.jandex.IndexView;
+import org.jboss.logging.Logger;
 
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
@@ -17,6 +25,18 @@ import io.quarkus.deployment.util.ServiceUtil;
  * {@link BuildStep}s related to {wsdl4j:wsdl4j}
  */
 class Wsdl4jProcessor {
+
+    private static final Logger log = Logger.getLogger(Wsdl4jProcessor.class);
+
+    @BuildStep
+    void indexDependencies(BuildProducer<IndexDependencyBuildItem> indexDependencies) {
+        Stream.of(
+                "wsdl4j:wsdl4j")
+                .forEach(ga -> {
+                    String[] coords = ga.split(":");
+                    indexDependencies.produce(new IndexDependencyBuildItem(coords[0], coords[1]));
+                });
+    }
 
     @BuildStep
     void httpProxies(BuildProducer<NativeImageProxyDefinitionBuildItem> proxies) {
@@ -30,7 +50,10 @@ class Wsdl4jProcessor {
     }
 
     @BuildStep
-    void reflectiveClass(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+    void reflectiveClass(
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        final IndexView index = combinedIndexBuildItem.getIndex();
 
         Stream.of("12", "")
                 .map(version -> ReflectiveClassBuildItem.builder(
@@ -44,6 +67,12 @@ class Wsdl4jProcessor {
 
         reflectiveClass.produce(ReflectiveClassBuildItem.builder("com.ibm.wsdl.extensions.schema.SchemaImpl").build());
 
+        final Set<String> extensibilityElements = index.getAllKnownImplementations(ExtensibilityElement.class)
+                .stream()
+                .map(classInfo -> classInfo.name().toString())
+                .collect(Collectors.toSet());
+        log.debugf("Registring ExtensibilityElements for reflection: %s", extensibilityElements);
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(extensibilityElements.toArray(new String[0])).build());
     }
 
     @BuildStep

--- a/integration-tests/client-server/pom.xml
+++ b/integration-tests/client-server/pom.xml
@@ -69,6 +69,24 @@
                 </executions>
             </plugin>
             <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>build-helper-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>add-source</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>add-source</goal>
+                        </goals>
+                        <configuration>
+                            <sources>
+                                <source>target/generated-sources/wsdl2java-extensors</source>
+                            </sources>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-resources-plugin</artifactId>
                 <executions>

--- a/integration-tests/client-server/src/main/java/io/quarkiverse/cxf/it/extensors/ExtensorsRest.java
+++ b/integration-tests/client-server/src/main/java/io/quarkiverse/cxf/it/extensors/ExtensorsRest.java
@@ -1,0 +1,25 @@
+package io.quarkiverse.cxf.it.extensors;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+import io.quarkiverse.cxf.annotation.CXFClient;
+import io.quarkiverse.cxf.it.server.extensors.model.ExtensorsService;
+
+@Path("/ExtensorsRest")
+public class ExtensorsRest {
+
+    @CXFClient("extensors")
+    ExtensorsService extensors;
+
+    @Path("/hello/{name}")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello(@PathParam("name") String name) {
+        return extensors.hello(name);
+    }
+
+}

--- a/integration-tests/client-server/src/main/java/io/quarkiverse/cxf/it/extensors/ExtensorsServiceImpl.java
+++ b/integration-tests/client-server/src/main/java/io/quarkiverse/cxf/it/extensors/ExtensorsServiceImpl.java
@@ -1,0 +1,17 @@
+package io.quarkiverse.cxf.it.extensors;
+
+import jakarta.jws.WebService;
+
+import io.quarkiverse.cxf.annotation.CXFEndpoint;
+import io.quarkiverse.cxf.it.server.extensors.model.ExtensorsService;
+
+@WebService(targetNamespace = "http://test.deployment.cxf.quarkiverse.io/", name = "ExtensorsService")
+@CXFEndpoint("/ExtensorsService")
+public class ExtensorsServiceImpl implements ExtensorsService {
+
+    @Override
+    public String hello(String name) {
+        return "Hello " + name;
+    }
+
+}

--- a/integration-tests/client-server/src/main/resources/application.properties
+++ b/integration-tests/client-server/src/main/resources/application.properties
@@ -218,4 +218,13 @@ quarkus.cxf.client.javaNetSslClient.client-endpoint-url = https://localhost:${qu
 quarkus.cxf.client.javaNetSslClient.service-interface = io.quarkiverse.cxf.it.HelloService
 quarkus.cxf.client.javaNetSslClient.logging.enabled = true
 
+# ExtensorsTest
+quarkus.cxf.codegen.wsdl2java.extensors.includes = wsdl/ExtensorsService.wsdl
+quarkus.cxf.codegen.wsdl2java.extensors.package-names = io.quarkiverse.cxf.it.server.extensors.model
+quarkus.cxf.codegen.wsdl2java.extensors.wsdl-location = classpath:wsdl/ExtensorsService.wsdl
+quarkus.cxf.codegen.wsdl2java.extensors.output-directory = target/generated-sources/wsdl2java-extensors
+
+quarkus.cxf.client.extensors.wsdl=wsdl/ExtensorsService.wsdl
+quarkus.cxf.client.extensors.client-endpoint-url=http://localhost:${quarkus.http.test-port}/soap/ExtensorsService
+
 quarkus.default-locale = en_US

--- a/integration-tests/client-server/src/main/resources/wsdl/ExtensorsService.wsdl
+++ b/integration-tests/client-server/src/main/resources/wsdl/ExtensorsService.wsdl
@@ -1,0 +1,68 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<wsdl:definitions
+    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+    xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/"
+    xmlns:tns="http://test.deployment.cxf.quarkiverse.io/"
+    xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/"
+    xmlns:ns1="http://schemas.xmlsoap.org/soap/http"
+    name="ExtensorsService"
+    targetNamespace="http://test.deployment.cxf.quarkiverse.io/"
+    xmlns:http="http://schemas.xmlsoap.org/wsdl/http/" >
+  <wsdl:types>
+<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:tns="http://test.deployment.cxf.quarkiverse.io/" attributeFormDefault="unqualified" elementFormDefault="unqualified" targetNamespace="http://test.deployment.cxf.quarkiverse.io/">
+  <!-- my custom comment -->
+  <xsd:element name="hello" type="tns:hello"/>
+  <xsd:complexType name="hello">
+    <xsd:sequence>
+      <xsd:element minOccurs="0" name="helloName" type="xsd:string"/>
+    </xsd:sequence>
+  </xsd:complexType>
+  <xsd:element name="helloResponse" type="tns:helloResponse"/>
+  <xsd:complexType name="helloResponse">
+    <xsd:sequence>
+      <xsd:element minOccurs="0" name="return" type="xsd:string"/>
+    </xsd:sequence>
+  </xsd:complexType>
+</xsd:schema>
+  </wsdl:types>
+  <wsdl:message name="helloResponse">
+    <wsdl:part element="tns:helloResponse" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:message name="hello">
+    <wsdl:part element="tns:hello" name="parameters">
+    </wsdl:part>
+  </wsdl:message>
+  <wsdl:portType name="ExtensorsService">
+    <wsdl:operation name="hello">
+      <wsdl:input message="tns:hello" name="hello">
+    </wsdl:input>
+      <wsdl:output message="tns:helloResponse" name="helloResponse">
+    </wsdl:output>
+    </wsdl:operation>
+  </wsdl:portType>
+  <wsdl:portType name="ExtensorsServiceGet" />
+  <wsdl:binding name="ExtensorsServiceSoapBinding" type="tns:ExtensorsService">
+    <soap:binding style="document" transport="http://schemas.xmlsoap.org/soap/http"/>
+    <wsdl:operation name="hello">
+      <soap:operation soapAction="" style="document"/>
+      <wsdl:input name="hello">
+        <soap:body use="literal"/>
+      </wsdl:input>
+      <wsdl:output name="helloResponse">
+        <soap:body use="literal"/>
+      </wsdl:output>
+    </wsdl:operation>
+  </wsdl:binding>
+  <wsdl:binding name="ExtensorsServiceGet" type="tns:ExtensorsServiceGet">
+    <http:binding verb="GET" />
+  </wsdl:binding>
+  <wsdl:service name="ExtensorsService">
+    <wsdl:port binding="tns:ExtensorsServiceSoapBinding" name="ExtensorsServicePort">
+      <soap:address location="http://localhost:8081/soap/ExtensorsService/"/>
+    </wsdl:port>
+    <wsdl:port binding="tns:ExtensorsServiceGet" name="ExtensorsServiceGet" >
+      <http:address location="http://localhost:8081/soap/ExtensorsServiceGet" />
+    </wsdl:port>
+  </wsdl:service>
+</wsdl:definitions>

--- a/integration-tests/client-server/src/test/java/io/quarkiverse/cxf/it/extensors/ExtensorsTest.java
+++ b/integration-tests/client-server/src/test/java/io/quarkiverse/cxf/it/extensors/ExtensorsTest.java
@@ -1,0 +1,21 @@
+package io.quarkiverse.cxf.it.extensors;
+
+import java.util.concurrent.ExecutionException;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+
+@QuarkusTest
+public class ExtensorsTest {
+    @Test
+    void consultarRestringido() throws InterruptedException, ExecutionException {
+        RestAssured.given()
+                .get("/ExtensorsRest/hello/Aureliano Buendía")
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("Hello Aureliano Buendía"));
+    }
+}

--- a/integration-tests/client-server/src/test/java/io/quarkiverse/cxf/it/extensors/ExtensorsTestIT.java
+++ b/integration-tests/client-server/src/test/java/io/quarkiverse/cxf/it/extensors/ExtensorsTestIT.java
@@ -1,0 +1,8 @@
+package io.quarkiverse.cxf.it.extensors;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+
+@QuarkusIntegrationTest
+public class ExtensorsTestIT extends ExtensorsTest {
+
+}

